frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -257,16 +264,18 @@ class FieldValuesWidgetInner extends Component {
       disablePKRemappingForSearch,
       loadingState,
       options,
+      hasIncompleteValueSet,
     });
 
     const isLoading = loadingState === "LOADING";
     const isFetchingList =
       shouldList(this.props.fields, this.props.disableSearch) && isLoading;
-    const hasListData = hasList({
-      fields,
-      disableSearch,
-      options,
-    });
+    const hasExhaustedListData =
```

Comment:
`hasExhaustedListData` (previously `hasListData`) is the boolean we use to determine if we are using `<ListField>` or `<TokenField>` to display the list of values. I've tweaked this boolean to depend on `hasIncompleteValueSet` because `<ListField>` does not contain any logic to handle searching for values via endpoint calls.

I'd prefer we **don't** duplicate fetching logic inside of `<ListField>`, as this component is already plenty gross and I wouldn't want that grossness to encroach on more components. Instead, since this is an edge case, I'm rendering `TokenField` again so that all field values become searchable.